crates/miden-lib/asm/miden/contracts/auth/multisig.masm

```diff
@@ -1,24 +1,208 @@
 use.miden::account
 use.miden::tx
+use.miden::auth
+
 use.std::crypto::dsa::rpo_falcon512
+use.std::crypto::hashes::rpo
+use.std::sys
 
 # CONSTANTS
 # =================================================================================================
 
 # Event to place the falcon signature of a provided message and public key on the advice stack.
-const.FALCON_SIG_TO_STACK=131087
+const.AUTH_REQUEST=131087
+const.UNAUTHORIZED_EVENT=131100
+
+const.THRESHOLD_SLOT=0
+const.NUMBER_OF_APPROVERS_SLOT=1
+const.PUBLIC_KEYS_MAP_SLOT=2
+const.EXECUTED_TXS_SLOT=3
+
+# CONSTANTS
+# AdviceMap Keys
+
+# Runtime Memory Addresses
+const.LOOP_ITER=100
+const.TEMP_PK=108
+const.MESSAGE=112
+const.THRESHOLD_VALUE=116
+const.OWNER_PUB_KEY=120
+
+# ERRORS
+const.NON_CONSTANT_MSG="NON CONSTANT SIGNATURES"
+const.INVALID_SIGNATURE="SIG DOES NOT MATCH DIGEST"
+
+# => []
+
+proc.verify_signatures_loop
+    push.0
+    # => [initial_iteration_value]
+
+    mem_store.LOOP_ITER
+    # => []
+
+    push.1
+    while.true
+        mem_load.LOOP_ITER
+        # => [iteration]
```

Comment:
https://github.com/0xMiden/miden-base/blob/858c998950288157ffb167606a898ce0b70d6a71/crates/miden-lib/asm/account_components/rpo_falcon_512_procedure_acl.masm#L48

here's a way to do the loop w/o using memory for the counters